### PR TITLE
Fetch assets saga

### PR DESCRIPTION
**Pull request by FrederikBolding**

### CODE REVIEW

**Comment by FrederikBolding** [comment]
[ch7317]

File: src/services/Store/store/asset.slice.ts [hunk 1]
```diff
@@ -67,3 +72,20 @@ export const {
...
+export function* assetSaga() {
+  yield all([
+    takeLatest(fetchAssets.type, fetchAssetsWorker),
+    // Trigger fetching assets on RESET, is dispatched when resetting and importing new settings
+    takeLatest(ActionT.RESET, fetchAssetsWorker)
+  ]);
+}
+
+export function* fetchAssetsWorker() {
+  const fetchedAssets = yield call(MyCryptoApiService.instance.getAssets);
```
Comment (vilsbole) [inline src/services/Store/store/asset.slice.ts:88]: Let's place our api calls inside a try/catch and add dispatch an error action to an Error reducer.
1. It anticipates centralised error handling.
2. It avoids errors bubbling to react.

File: src/services/Store/store/asset.slice.ts [hunk 2]
```diff
@@ -67,3 +72,20 @@ export const {
...
 export default slice;
+
+/**
+ * Sagas
+ */
+export function* assetSaga() {
+  yield all([
+    takeLatest(fetchAssets.type, fetchAssetsWorker),
+    // Trigger fetching assets on RESET, is dispatched when resetting and importing new settings
+    takeLatest(ActionT.RESET, fetchAssetsWorker)
```
Comment (vilsbole) [inline src/services/Store/store/asset.slice.ts:83]: 😻 Wunderbar!

File: src/services/Store/store/asset.slice.test.ts [hunk 3]
```diff
@@ -67,3 +73,16 @@ describe('AccountSlice', () => {
     expect(actual).toEqual(initialState);
   });
 });
+
+describe('fetchAssetsWorker()', () => {
+  expectSaga.DEFAULT_TIMEOUT = 100;
```
Comment (vilsbole) [inline src/services/Store/store/asset.slice.test.ts:78]: I guess it is useful for all our saga tests.
In this case can we set the default in `test-utils` and import from there ?

**Review by vilsbole — CHANGES_REQUESTED** [review]

**Review by vilsbole — APPROVED** [review]

**Pull request merged**